packages/backend/src/index.ts

```diff
@@ -432,6 +434,24 @@ export function initIocContainer(
     })
   })
 
+  container.singleton('ilpPaymentService', async (deps) => {
+    return createIlpPaymentService({
+      logger: await deps.use('logger'),
+      knex: await deps.use('knex'),
+      config: await deps.use('config'),
+      makeIlpPlugin: await deps.use('makeIlpPlugin'),
+      ratesService: await deps.use('ratesService')
+    })
+  })
+
+  container.singleton('paymentMethodHandlerService', async (deps) => {
+    return createPaymentMethodHandlerService({
+      logger: await deps.use('logger'),
+      knex: await deps.use('knex'),
+      ilpPaymentService: await deps.use('ilpPaymentService')
+    })
+  })
```

Comment:
To try and build on this idea, maybe it could take in an object whose keys are the names of supported payment methods, and the values are its services. So `getQuote` would then look like:
```
/*
  PaymentMethods: {
    ilp: IlpPaymentService
    ...other future payment method services
  }

  paymentMethods: PaymentMethods
*/

getQuote: (method, quoteOptions) => paymentMethods[method].getQuote(quoteOptions)

Reply:
@raducristianpopa the `PaymentMethodHandlerService` is our "factory" in this case already, but instead of a new object instance we're just returning the service/singleton itself

@njlie yeah I think I'll change it to the suggestion since the types make it (more) "safe" and prevent passing in an invalid payment method  :+1: